common/ext/class.GenerisInstaller.php

```diff
@@ -46,12 +50,24 @@ public function install()
         }
  
         $this->installLoadDefaultConfig();
+        $this->setupSchema();
         $this->installOntology();
         $this->installRegisterExt();
-        
+
         common_cache_FileCache::singleton()->purge();
         
         $this->log('d', 'Installing custom script for extension ' . $this->extension->getId());
         $this->installCustomScript();
     }
+
+    public function setupSchema()
+    {
+        $pm = $this->getServiceManager()->get(PersistenceManager::SERVICE_ID);
+        $schemaCollection = $pm->getSqlSchemas();
+        $ontology = $this->getServiceManager()->get(Ontology::SERVICE_ID);
+        if ($ontology instanceof SchemaAwareInterface) {
```

Comment:
Unit test?

Reply:
Install process currently not unit testable, and would require a major refactoring